cmd/builder/internal/builder/main.go

```diff
@@ -4,28 +4,65 @@
 package builder // import "go.opentelemetry.io/collector/cmd/builder/internal/builder"
 
 import (
+	"bytes"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strings"
 	"text/template"
 	"time"
 
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapio"
+
+	"go.opentelemetry.io/collector/cmd/builder/internal/builder/modfile"
 )
 
 var (
 	// ErrGoNotFound is returned when a Go binary hasn't been found
-	ErrGoNotFound = errors.New("go binary not found")
+	ErrGoNotFound       = errors.New("go binary not found")
+	ErrStrictMode       = errors.New("failing due to strict mode")
+	errFailedToDownload = errors.New("failed to download go modules")
 )
 
-func runGoCommand(cfg Config, args ...string) error {
-	cfg.Logger.Info("Running go subcommand.", zap.Any("arguments", args))
+// makeGoCommand is called by runGoCommand and runGoCommandStdout, mainly
+// to isolate the security annotation.
+func makeGoCommand(cfg Config, args []string) *exec.Cmd {
 	// #nosec G204 -- cfg.Distribution.Go is trusted to be a safe path and the caller is assumed to have carried out necessary input validation
 	cmd := exec.Command(cfg.Distribution.Go, args...)
 	cmd.Dir = cfg.Distribution.OutputPath
+	return cmd
+}
+
+// runGoCommandStdout is similar to `runGoCommand` but returns the
+// standard output.  The subcommand is only printed with Verbose=true,
+// since it is invoked frequently when --skip-new-go-module is set.
```

Comment:
I think this comment is not relevant until #9631. I would just change `runGoCommand` for now and make it return the standard output if needed